cg/meta/archive/archive.py

```diff
@@ -20,24 +39,24 @@ def __init__(self, housekeeper_api: HousekeeperAPI, status_db: Store):
         self.housekeeper_api: HousekeeperAPI = housekeeper_api
         self.status_db: Store = status_db
 
-    def get_files_by_archive_location(
-        self, file_data: List[PathAndSample], archive_location: ArchiveLocationsInUse
-    ) -> List[PathAndSample]:
-        """
-        Returns a list of PathAndSample where the associated sample has a specific archive location.
-        """
-        selected_files: List[PathAndSample] = []
-        for file in file_data:
-            sample: Optional[Sample] = self.get_sample(file)
-            if sample and sample.archive_location == archive_location:
-                selected_files.append(file)
-        return selected_files
-
-    def get_sample(self, file: PathAndSample) -> Optional[Sample]:
-        sample: Optional[Sample] = self.status_db.get_sample_by_internal_id(file.sample_internal_id)
+    def get_sample(self, file: File) -> Optional[Sample]:
+        """Fetches the Sample corresponding to a File and logs if a Sample is not found."""
+        sample: Optional[Sample] = self.status_db.get_sample_by_internal_id(
+            file.version.bundle.name
+        )
         if not sample:
             LOG.warning(
```

Comment:
Out of scope, but is it appropriate that this is a warning? If we expect it to happen in some instances, maybe it should be an info or debug log

Reply:
If I am not mistaken, all spring files in Housekeeper should be added to a bundle which corresponds to a sample in status_db. So if no such sample is found, it signifies some issue with the sample. We probably do not want to raise an error since we are running this in batches, though, so archiving the remaining files would still be wished for.